### DISCUSSION TITLE
Using reverse-proxy authentication?

### DISCUSSION THREAD

**Discussion by gardiol** [joplin-vieweb/joplin-vieweb#69] [Q&A]
Hi!
i have an extensive setup with many services and whatnot, all behind a NGINX reverse proxy. I have NGINX doing the authentication for many services that support passtrough from reverse proxy authentication. Is it possible to achieve this also using this amazing piece of software?

This would avoid users to double authenticate (once to access the subdomain, once to access the service).

Many services (lke PodFetch of GITEA for example) support this neat feature.

Is it already availble and i missed it or could it be added (no rush!) otherwise?

Usually, i have to configure in the reverse proxy some specific header name that carrier the username for this to work.

**Comment by gri38** [comment]
Hello @gardiol .

Good idea !  
Currently it's not possible, but it shouldn't be hard to add a setup option. I'll have a look and let you know in this discussion.

**Comment by gri38** [comment]
Hello @gardiol 
See https://github.com/joplin-vieweb/joplin-vieweb/releases/tag/v2.6.0

Let me know if it suits you.

**Comment by gardiol** [comment]
Looks interesting, and working. Two questions:

- this will not cope with multiple users, correct?
- in case of a fresh install, how will this map the user? Will it match a user in django with the reverse proxy user?: well, **just tinsert synchronization backed accordingly!**
- i didnt test yet, but if i access without the proxy auth, will this still work or not? 

Thank s for the really fast fix! I love Joplin Vieweb :)

**Comment by gri38** [comment]
Hi
- Today joplin vieweb cannot work with multiple users / accoutn, because joplin terminal doesn't.
So, if I understand well your question: this topic doesn't handle multiple joplin users. And it doesn't support multiple proxy auth: joplin just displays all the notes, and your proxy grant access to logged users... but every users will have the same notes.
- so, no mapping between django users and proxy users: if you configure JOPLIN_LOGIN_REQUIRED to False, joplin just displays notes.
  - when JOPLIN_LOGIN_REQUIRED is True, joplin requires a logged in user, but any user will see the same notes...
- probably, but without any auth security

**Comment by gardiol** [comment]
thank you! Everything is perfectly clear! I think you solved my original issue. Thanks!
